packages/preview/cqut-thesis-typst/0.1.0/utils/theorem.typ

```diff
@@ -0,0 +1,165 @@
+#import "@preview/ctheorems:1.1.2": *
+#import "@preview/showybox:2.0.1": showybox
+#import "../utils/style.typ": 字号, 字体
+
+#let colors = (
+  rgb("#9E9E9E"),
+  rgb("#F44336"),
+  rgb("#E91E63"),
+  rgb("#9C27B0"),
+  rgb("#673AB7"),
+  rgb("#3F51B5"),
+  rgb("#2196F3"),
+  rgb("#03A9F4"),
+  rgb("#00BCD4"),
+  rgb("#009688"),
+  rgb("#4CAF50"),
+  rgb("#8BC34A"),
+  rgb("#CDDC39"),
+  rgb("#FFEB3B"),
+  rgb("#FFC107"),
+  rgb("#FF9800"),
+  rgb("#FF5722"),
+  rgb("#795548"),
+  rgb("#9E9E9E"),
+)
+
+
+#show: thmrules
+
+
+
+#show math.equation: set text(weight: 400)
+
+#let thmtitle(t, color: rgb("#000000")) = {
+  return text(font: 字体.黑体, weight: "semibold", fill: color)[#t]
+}
+#let thmname(t, color: rgb("#000000")) = {
+  return text(font: 字体.黑体, fill: color)[(#t)]
+}
+
+#let thmtext(t, color: rgb("#000000")) = {
+  let a = t.children
+  if (a.at(0) == [ ]) {
+    a.remove(0)
+  }
+  t = a.join()
+
+  return text(font: 字体.宋体, fill: color)[#t]
+}
+
+#let thmbase(
+  identifier,
+  head,
+  ..blockargs,
+  supplement: auto,
+  padding: (top: 0.5em, bottom: 0.5em),
+  namefmt: x => [(#x)],
+  titlefmt: strong,
+  bodyfmt: x => x,
+  separator: [#h(0.1em)\ ], 
+  base: "heading",
+  base_level: none,
```

Comment:
To be more coherent with the rest of the package (and with Typst conventions in general) it would be nice to rename this to `base-level`.